core/src/main/java/org/apache/iceberg/util/ZOrderByteUtils.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
How does this relate to https://github.com/apache/iceberg/issues/3960?

Reply:
I see comment below about splitting it out.

Reply:
Yep sorry, this pr is built on top of 3960 so hopefully we can get things in faster.